### PR TITLE
Add functional support of AB group scale

### PR DESCRIPTION
**Pull request by zjing14**
- Added grid/device ops for AB group scale
- Added an example for fp8

### CODE REVIEW

**Comment by jianyuh** [comment]
CC @jwfromm

**Review by carlushuang — DISMISSED** [review]
LGTM

File: include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_cshuffle_v3_multi_d_ab_scale.hpp [hunk 1]
```diff
@@ -0,0 +1,2300 @@
...
+          bool HasMainKBlockLoop,
+          InMemoryDataOperationEnum CGlobalMemoryDataOperation,
+          index_t MinimumOccupancy = 1,
+          TailNumber TailNum       = TailNumber::Full>
+__global__ void
+#if CK_USE_LAUNCH_BOUNDS
+    __launch_bounds__(CK_MAX_THREAD_PER_BLOCK, MinimumOccupancy)
+#endif
+    // __attribute__((amdgpu_waves_per_eu(1, 1)))
+    kernel_gemm_xdl_cshuffle_v3(typename GridwiseGemm::Argument karg)
```
Comment (jwfromm) [inline include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_cshuffle_v3_multi_d_ab_scale.hpp:39]: When building in a larger project, the name of this function ends up conflicting with other similarly defined kernels. Consider giving this a unique name.

File: example/65_gemm_multiply_multiply/gemm_multiply_multiply_xdl_fp8_ab_scale.cpp [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+    ck::index_t StrideA = K;
+    ck::index_t StrideB = K;
+    ck::index_t StrideE = N;
+
+    ck::index_t Scale_Block_M = 256;
+    ck::index_t Scale_Block_N = 256;
+    ck::index_t Scale_Block_K = 256;
+
+    ck::index_t Scale_Stride_AM = Scale_Block_K;
```
Comment (jwfromm) [inline example/65_gemm_multiply_multiply/gemm_multiply_multiply_xdl_fp8_ab_scale.cpp]: Should this `K / 256`? I would expect that block size means how many values along that axis share a value. In this example it instead appears to mean the size of the scale block.

Comment (zjing14) [reply example/65_gemm_multiply_multiply/gemm_multiply_multiply_xdl_fp8_ab_scale.cpp]: No, it is not required. But Scale_Block_K need be multiple of KPerBlock.

File: include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_cshuffle_v3_multi_d_ab_scale.hpp (hunk 1 above)
Comment (zjing14) [reply include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_cshuffle_v3_multi_d_ab_scale.hpp:39]: Will fix it.

**Comment by aska-0096** [comment]
@jwfromm I'm checking into this thread, will review the code.

**Comment by aska-0096** [comment]
Close this PR.
@jwfromm draft PR #1384 created, including everything related to f8 gemm you need.

**Pull request closed without merging**